server/api/lib/trees/hod_hasharon_tree_permit.js

```diff
@@ -0,0 +1,153 @@
+const hodHashTreePermit = {
+  urls: [TREES_HOD_HASHARON_URL]
+};
+
+async function parseTreesHtml(url) {
+  const treesHtml = await proxy.get(url);
+  const dom = cheerio.load(treesHtml, {
+    decodeEntities: false
+  });
+  if (!dom) {
+    console.error('cheerio dom is null');
+  }
+  const keys = [];
+  const result = [];
+  dom('.page-content-body').find('table').find('tr').each((row, elem) => {
+    if (row === 0) {
+      dom(elem).find('th').each((idx, elem) => {
+        const key = dom(elem).text().trim();
+        keys.push(key);
+      });
+      return;
+    }
+    const treePermit = {};
+    dom(elem).find('td,th').each((idx, elem) => {
+            const url = idx === 6 ? dom(elem).find('a').attr('href') : '';
+            if (url.length > 0) {
+                treePermit['url'] = url.substr(1);
+            }
+      const value = dom(elem).text().trim();
+      const key = keys[idx];
+      treePermit[key] = value;
+    });
+    result.push(treePermit);
+  });
+  Log.info(`number of hod hasharon permits: ${result.length}`);
+  return result;
+}
+
+function processRawPermits(rawPermits) {
+  try {
+    const treePermits = rawPermits.map(raw => {
+      try{        
+        const last_date_to_objection = formatDate(raw[OBJECTION_TILL], '09:00', 'DD.MM.YY');
+        if (!last_date_to_objection) {
+          Log.error(`No / Bad dates format, ignore this license: hod hasharon, ${raw[STREET_NAME]} , ${raw[OBJECTION_TILL]}`);
+          return null;
+        }
+        
+        const treesPerPermit = parseTreesPerPermit(raw[TREE_TYPE], raw[TREE_NUM]);
+        const totalTrees = sum(treesPerPermit);
+        
+        const permitNumber = `meirim-hodash-${raw[LICENSE_NUMBER]}`;
+
+        const attributes = {
+          [REGIONAL_OFFICE]: CITY,
+          [PLACE]: CITY,
+          [APPROVER_TITLE]: APPROVER,
+          [PERMIT_NUMBER]: permitNumber,
+          [STREET]: raw[STREET_NAME],
+          [ACTION]: SHORT_ACTION,
+          [LAST_DATE_TO_OBJECTION]: last_date_to_objection,
+          [PERSON_REQUEST_NAME]: raw[LICENSE_OWNER],
+          [REASON_SHORT]: raw[LICENSE_REASON],
+          [TREES_PER_PERMIT]: treesPerPermit,
+          [TOTAL_TREES]: totalTrees,
+          
+          [TREE_PERMIT_URL]: 'https://www.hod-hasharon.muni.il' + raw['url'],
+        };
+        const permit = new TreePermit(attributes);
+        return permit;
+      }
+      catch (e) {
+        Log.error(`error in hod hasharon parse row, ignoring: ${raw[STREET_NAME]}`, e.message);
+        return null;
+      }
+    }
+    );
+    return treePermits.filter(Boolean); // remove undefined values;
+  }
+  catch (e) {
+    Log.error('error in hod hasharon parse rows:' + e);
+  }
+}
+
+function parseTreesPerPermit(treesInPermitStr, treeAmount) {
+    treesInPermitStr = cleanString(treesInPermitStr);
+    treeAmount = cleanString(treeAmount);
+    const linesName = treesInPermitStr.split('\n');
+    const linesAmount = treeAmount.split('\n');
```

Comment:
Why not add these lines to `cleanString`?

Reply:
its not cleaning the string
got a new name in mind?

Reply:
done